diagnostic.js

```diff
@@ -29,26 +29,64 @@ mongoose.Promise = global.Promise;
 mongoose.connect('mongodb://localhost/mongoose-crud');
 
 /// ADD YOUR CODE BELOW
+const Movement = require('./models/movement.js');
 
-const create = (name, description, startYear, endYear) => {};
+const db = mongoose.connection;
+
+const done = function() {
+  db.close();
+};
+
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+  name: name,
+  description: description,
+  startYear: startYear,
+  endYear: endYear
+}).then(function(place) {
+  console.log(place.toJSON());
+}).catch(function(err) {
+  console.error(err);
+}).then(done);
+};
 // Success -> Print new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  Movement.find({}).then(function(people){
+  people.forEach(function(person){
+    console.log(person.toJSON());
+  });
+}).catch(console.error).then(done);
+};
 // Success -> Print all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id).then(function(person){
+console.log(person.toJSON());
+}).catch(console.err).then(done);
+};
```

Comment:
Close, but you forgot one thing - not finding something is not the same as having an error!
As a result, you need to check to see if `movement` exists before printing it out; if it doesn't exist, you were directed to print out "Not Found" and exit (analogous to returning a 404 in Express).

Also, that should probably be `movement`, not `person`, right?